go/filemanager/pfsmodules/download.go

```diff
@@ -1,137 +1,65 @@
 package pfsmodules
 
 import (
-	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
-	"mime/multipart"
 	"os"
 	"path/filepath"
 
-	"github.com/PaddlePaddle/cloud/go/utils/restclient"
 	log "github.com/golang/glog"
 )
 
-func remoteChunkMeta(path string,
-	chunkSize int64) ([]ChunkMeta, error) {
-	cmd := ChunkMetaCmd{
-		Method:    ChunkMetaCmdName,
-		FilePath:  path,
-		ChunkSize: chunkSize,
-	}
-
-	t := fmt.Sprintf("%s/api/v1/pfs/chunks", Config.ActiveConfig.Endpoint)
-	ret, err := restclient.GetCall(t, cmd.ToURLParam())
-	if err != nil {
-		return nil, err
-	}
-
-	type chunkMetaResponse struct {
-		Err     string      `json:"err"`
-		Results []ChunkMeta `json:"results"`
-	}
-
-	resp := chunkMetaResponse{}
-	if err := json.Unmarshal(ret, &resp); err != nil {
-		return nil, err
-	}
-
-	if len(resp.Err) == 0 {
-		return resp.Results, nil
+func downloadFile(src string, srcFileSize int64, dst string) error {
+	w := FileHandle{}
+	if err := w.Open(dst, os.O_RDWR, srcFileSize); err != nil {
+		return err
 	}
+	defer w.Close()
 
-	return resp.Results, errors.New(resp.Err)
-}
-
-func getChunkData(target string, chunk Chunk, dst string) error {
-	log.V(1).Info("target url: " + target)
-
-	resp, err := restclient.GetChunk(target, chunk.ToURLParam())
-	if err != nil {
+	r := RFileHandle{}
+	if err := r.Open(src, os.O_RDONLY, 0); err != nil {
 		return err
 	}
-	defer Close(resp.Body)
+	defer r.Close()
 
-	if resp.Status != restclient.HTTPOK {
-		return errors.New("http server returned non-200 status: " + resp.Status)
-	}
+	offset := int64(0)
+	size := defaultChunkSize
 
-	partReader := multipart.NewReader(resp.Body, DefaultMultiPartBoundary)
 	for {
-		part, error := partReader.NextPart()
-		if error == io.EOF {
-			break
+		m, err := r.GetChunkMeta(offset, size)
```

Comment:
seems `m` and `c` are both calculated based on the file that is already on disk (already uploaded to server). But I thought the purpose of chunking is to avoid uploading file to server?